runner.go

```diff
@@ -62,15 +80,29 @@ type doneInfo struct {
 // The function isFatal(err) returns whether err is a fatal error.  The
 // function moreImportant(err0, err1) returns whether err0 is considered
 // more important than err1.
-func NewRunner(isFatal func(error) bool, moreImportant func(err0, err1 error) bool, restartDelay time.Duration) Runner {
-	runner := &runner{
+func NewRunner(p RunnerParams) *Runner {
+	if p.IsFatal == nil {
+		p.IsFatal = func(error) bool {
+			return true
+		}
+	}
+	if p.MoreImportant == nil {
+		p.MoreImportant = func(err0, err1 error) bool {
+			return true
+		}
+	}
+	if p.RestartDelay == 0 {
+		p.RestartDelay = RestartDelay
+	}
+
+	runner := &Runner{
```

Comment:
Why not make RunnerParams part of this structure so you only have to assign one thing.